internal/pkg/service/stream/storage/metacleanup/metacleanup.go

```diff
@@ -176,10 +173,24 @@ func (n *Node) cleanMetadata(ctx context.Context) (err error) {
 		ListAll().
 		ForEach(func(job keboolaBridgeModel.Job, _ *iterator.Header) error {
```

Comment:
Why don't we put this whole block into the bridge as I suggested? :thinking: I'd like to avoid using `keboolaBridgeRepository` here in `metacleanup`.

Reply:
You will face the thing, that part of cleaning is spans/traces inside `bridge` which missleads, because real Process/component that reports the cleanup is `Metadata` component.
The most important is this one 
```
// There can be several cleanup nodes, each node processes an own part.
if !n.dist.MustCheckIsOwner(job.ProjectID.String()) {
	return nil
}
```
This says that there is only 1 coordinator node that cleans up the metadata, so there are not race conditions or other issues due to duplicate cleanups. This cannot be done on bridge level because it is not a process but a library.

Reply:
I see. I'll see if I can improve this using range over functions but that can be done separately. :)